src/views/exportJarFileUtil.ts

```diff
@@ -0,0 +1,336 @@
+                resolve(pickBox.selectedItems[0].uri);
+                pickBox.dispose();
+            });
+            pickBox.onDidHide(() => {
+                reject();
+                pickBox.dispose();
+            });
+            pickBox.show();
+        } else {
+            return reject("No workspace folder found.");
+        }
+    });
+}
+
+function generateJar(progress, token: CancellationToken, pickSteps: string[], rootNodes: INodeData[],
+                     description: string, outputPath: string): Promise<string | undefined> {
+    return new Promise<string | undefined>(async (resolve, reject) => {
+        if (token.isCancellationRequested) {
+            return reject("User Cancelled.");
+        } else if (rootNodes === undefined) {
+            return reject("No project found.");
+        }
+        progress.report({ increment: 10, message: "Resolving classpaths..." });
+        let outClassPaths: string[];
+        try {
+            outClassPaths = await generateOutClassPath(pickSteps, rootNodes, outputPath);
+        } catch (e) {
+            return reject(e);
+        }
+        const outputFileName = join(outputPath, basename(outputPath) + ".jar");
+        progress.report({ increment: 30, message: "Generating jar..." });
+        const exportResult = await Jdtls.exportJar(basename(description), outClassPaths, outputFileName);
+        if (exportResult === true) {
+            resolve(outputFileName);
+        } else {
+            reject("Export jar failed.");
+        }
+    });
+}
+
+function resolveMainMethod(progress, token: CancellationToken, pickSteps: string[], projectPath: string): Promise<string | undefined> {
+    return new Promise<string | undefined>(async (resolve, reject) => {
+        if (token.isCancellationRequested) {
+            return reject("User Cancelled.");
+        }
+        progress.report({ increment: 10, message: "Resolving main classes..." });
+        if (mainMethods === undefined || mainMethods.length === 0) {
+            return resolve("");
+        }
+        progress.report({ increment: 30, message: "Determining main class..." });
+        const pickNodes: IJarQuickPickItem[] = [];
+        for (const mainMethod of mainMethods) {
+            if (Uri.file(mainMethod.path).fsPath.includes(projectPath)) {
+                const jarQuickPickItem: IJarQuickPickItem = {
+                    label: getName(mainMethod),
+                    description: mainMethod.name,
+                };
+                pickNodes.push(jarQuickPickItem);
+            }
+        }
+        if (pickNodes.length === 0) {
+            return resolve("");
+        } else {
+            const pickBox = window.createQuickPick<IJarQuickPickItem>();
+            const noMainClassItem: IJarQuickPickItem = {
+                label: "No main class",
+                description: "",
+            };
+            pickNodes.push(noMainClassItem);
+            pickBox.items = pickNodes;
```

Comment:
The operation for the pickBox in different steps are very similar. Considering extract to a method?